src/main/java/frc/trigon/robot/subsystems/climber/Climber.java

```diff
@@ -0,0 +1,130 @@
+            0
+    ).withSlot(ClimberConstants.GROUNDED_SLOT).withEnableFOC(ClimberConstants.ENABLE_FOC),
+            onChainPositionRequest = new DynamicMotionMagicVoltage(
+                    0,
+                    ClimberConstants.MAX_ON_CHAIN_VELOCITY,
+                    ClimberConstants.MAX_ON_CHAIN_ACCELERATION,
+                    0
+            ).withSlot(ClimberConstants.ON_CHAIN_SLOT).withEnableFOC(ClimberConstants.ENABLE_FOC);
+    private final VoltageOut voltageRequest = new VoltageOut(0).withEnableFOC(ClimberConstants.ENABLE_FOC);
+    private ClimberConstants.ClimberState targetState = ClimberConstants.ClimberState.RESTING;
+
+    public Climber() {
+        setName("Climber");
+    }
+
+    @Override
+    public void periodic() {
+        rightMotor.update();
+        leftMotor.update();
+        updateMechanisms();
+    }
+
+    @Override
+    public void drive(Measure<Voltage> voltageMeasure) {
+        driveRightMotor(voltageMeasure.plus(Units.Volts.of(calculateStayingInPlaceVoltage(rightMotor.getSignal(TalonFXSignal.POSITION), targetState.affectedByRobotWeight))));
+        driveLeftMotor(voltageMeasure.plus(Units.Volts.of(calculateStayingInPlaceVoltage(leftMotor.getSignal(TalonFXSignal.POSITION), targetState.affectedByRobotWeight))));
+    }
+
+    @Override
+    public void updateLog(SysIdRoutineLog log) {
+        log.motor("RightClimberMotor")
+                .linearPosition(Units.Meters.of(rightMotor.getSignal(TalonFXSignal.POSITION)))
+                .linearVelocity(Units.MetersPerSecond.of(rightMotor.getSignal(TalonFXSignal.VELOCITY)))
+                .voltage(Units.Volts.of(rightMotor.getSignal(TalonFXSignal.MOTOR_VOLTAGE) - calculateStayingInPlaceVoltage(rightMotor.getSignal(TalonFXSignal.POSITION), targetState.affectedByRobotWeight)));
+        log.motor("LeftClimberMotor")
+                .linearPosition(Units.Meters.of(leftMotor.getSignal(TalonFXSignal.POSITION)))
+                .linearVelocity(Units.MetersPerSecond.of(leftMotor.getSignal(TalonFXSignal.VELOCITY)))
+                .voltage(Units.Volts.of(leftMotor.getSignal(TalonFXSignal.MOTOR_VOLTAGE) - calculateStayingInPlaceVoltage(leftMotor.getSignal(TalonFXSignal.POSITION), targetState.affectedByRobotWeight)));
+    }
+
+    @Override
+    public void setBrake(boolean brake) {
+        rightMotor.setBrake(brake);
+        leftMotor.setBrake(brake);
+    }
+
+    @Override
+    public SysIdRoutine.Config getSysIdConfig() {
+        return ClimberConstants.SYSID_CONFIG;
+    }
+
+    @Override
+    public void stop() {
+        rightMotor.stopMotor();
+        leftMotor.stopMotor();
+    }
+
+    public boolean atTargetState() {
+        return Math.abs(toMeters(rightMotor.getSignal(TalonFXSignal.POSITION)) - targetState.positionMeters) < ClimberConstants.CLIMBER_TOLERANCE_METERS &&
+                Math.abs(toMeters(leftMotor.getSignal(TalonFXSignal.POSITION)) - targetState.positionMeters) < ClimberConstants.CLIMBER_TOLERANCE_METERS;
```

Comment:
Split this into atRightTargetState and atLeftTargetState and keep a full one as well